File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function rightSideView(root: TreeNode | null): number[] {
  
  if(!root)
    return [];
  
  let Q = new Array(),
        ans = new Array();
  
  Q.push(root);
  
  while(Q.length) {
    let currQ = new Array();
    let j = 0;
    
    while(j < Q.length) {
      const curr = Q[j];
      j++;
      
      if(j == Q.length)
        ans.push(curr.val);
      
      if(curr.left)
        currQ.push(curr.left);

      if(curr.right)
        currQ.push(curr.right);
    }
    
    Q = currQ;
  }
  
  return ans;
};

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        Map<Integer, Integer> dict = new HashMap<>();

        this.bfs(0, root, dict);
        
        return List.copyOf(dict.values());
    }
    
    private void bfs(int level, TreeNode root, Map<Integer, Integer> dict) {
        if(root == null)
            return;
        
        dict.put(level, root.val);
        this.bfs(level + 1, root.left, dict);
        this.bfs(level + 1, root.right, dict);
    }
}